components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -65,35 +68,44 @@
     /**
      * Returning instance of one of the implementation of SessionDataStore.
      *
-     * @return {@link RdbmsSingleEntrySessionDataStore} or {@link RedisSessionDataStore}
+     * @return {@link RdbmsSingleEntrySessionDataStore} or {@link RedisSessionDataStore} or {@link RdbmsMultiEntrySessionDataStore}
      */
     public static SessionDataStore getInstance() {
 
-        String enablePersistVal = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.Enable");
-        enablePersist = true;
-        if (enablePersistVal != null) {
-            enablePersist = Boolean.parseBoolean(enablePersistVal);
-        }
-
-        String redisEnabledVal = IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.RedisEnable");
-        if (StringUtils.isNotBlank(redisEnabledVal)) {
-            redisEnabled = Boolean.parseBoolean(redisEnabledVal);
-        }
         if (instance == null) {
             synchronized (SessionDataStore.class) {
                 if (instance == null) {
-                    if (redisEnabled == true) {
-                        instance = new RedisSessionDataStore();
+                    if (FrameworkServiceDataHolder.getInstance().getSessionDataStores().containsKey(getImplType())) {
+                        instance = FrameworkServiceDataHolder.getInstance().getSessionDataStores().get(getImplType());
                     } else {
-                        instance = new RdbmsSingleEntrySessionDataStore();
+                        instance = new RdbmsMultiEntrySessionDataStore();
                     }
```

Comment:
here we are creating the instance manually, but the issue is that creation is not a task of the session store. That is handled by the OSGi activate method.

You can,

1. Create a constant called `DEFAULT_SESSION_DATA_STORE` and make it equal to rdbms multu entry's name.
2. Then remove this else block. Set instance = DEFAULT_SESSION_DATA_STORE. Afterwards, put the `if` block